ide/libs.graalsdk/src/org/netbeans/libs/graalsdk/impl/GraalEnginesProvider.java

```diff
@@ -60,7 +61,9 @@ public List<ScriptEngineFactory> factories(ScriptEngineManager m) {
             if (disable == null) {
                 enumerateLanguages(arr, m == null ? null : m.getBindings());
             }
-        } catch (IllegalStateException | LinkageError err) {
+        } catch (IllegalStateException | LinkageError | ServiceConfigurationError err) {
+            // See NETBEANS #7245: First-time initialization with broken implementation may produce ServiceConfigurationError, As this
+            // happens in class-initializer, subsequent attempts will throw LinkageErrors.
```

Comment:
Thank you - indeed a more focused fix, bringing a partitial solution with minimal/no negative side effects.  At this point however I think it would be helpful to also have logging as in `Scripting.java`. Adding the catch clause for `ServiceConfigurationError` fixes the problem reported in #7245 (not being able to generate a HTML/JS application), but there is also no info shown, as the exception is swallowed. This also does not need an additional guard as the code that could cause the exceptions is entered at most once (`disable` is used as guard).

Reply:
OK to more logging here. I do not quite understand the part with `disable`: as far as I read the code, factories() can be called as part of `Scripting.newBuilder().build()`, each call would produce an additional exception (since Graal SDK is unable to enumerate its languages). The `disable` aimed to prevent that  and log Graal provider's failure just once.

Is it desirable to log each time Scripting.newBuilder().build() is called ?

Reply:
I meant, that we should log the exception here, as it is not reported anywhere else and relevant it a user for example complains about missing PAC support.

The `disable` part was only meant to say, that from my POV a plain log without an additional guard would be ok, because the exception code path will only be hit once. In `Scripting.java` you added `faultyLookupIds` to prevent duplicate reports and that is not necessary here.

Reply:
My understanding is, that `GraalEnginesProvider.java` is loaded as a singleton and thus the second call to `factories` will not call into `enumerateLanguages` because `disable` is already set.

Reply:
yeah this looks like the catch block can be only reached once since it writes to `disable` and nothing is ever resetting it -> logging here would be no problem

Reply:
@mbien & @matthiasblaesing : finally understood. thanks.